src/main/java/domain/GameBoard.java

```diff
@@ -1,18 +1,77 @@
 package domain;
 
 import domain.ladder.Ladder;
-import domain.ladder.common.Direction;
+import domain.ladder.attribute.Direction;
+import domain.common.Name;
+import domain.player.Player;
 import domain.player.Players;
+import domain.reward.Result;
+import domain.reward.Reward;
+import domain.reward.Rewards;
 
 import java.util.List;
+import java.util.stream.Stream;
 
 public class GameBoard {
+    private static final int COLUMN_START_POSITION = 0;
     private final Players players;
     private final Ladder ladder;
+    private final Rewards rewards;
 
-    public GameBoard(Players players, Ladder ladder) {
+    public GameBoard(Players players, Ladder ladder, Rewards rewards) {
         this.players = players;
         this.ladder = ladder;
+        this.rewards = rewards;
+    }
+
+    public Player findPlayerWithName(Name name) {
+        return players.getPlayerWithName(name);
+    }
+
+    public Result playGameOnePlayer(Player player) {
+        Point startPoint = getPlayerStartPoint(player);
+        Point endPoint = playGameWithStartPoint(startPoint);
+        return new Result(player.getName(), getRewardWithIndex(endPoint.row()));
+    }
+
+    public List<Result> playGameAllPlayer() {
+        return players.getPlayers()
+                      .stream()
+                      .map(this::playGameOnePlayer)
+                      .toList();
+    }
+
+    private Point playGameWithStartPoint(final Point startPoint) {
+
+        return Stream.iterate(startPoint, this::movePoint)
+                     .filter(this::isPointIsEndLine)
+                     .findFirst()
+                     .orElseThrow(() -> new IllegalStateException("""
+                             만족하는 결과가 없는 경우로 , 사다리가 잘못 생성되었습니다.
+                             사다리를 다시 생성해주세요!
+                             """));
+    }
+
+    private Point movePoint(Point point) {
+        Direction direction = ladder.getDirectionWithRowAndColumn(point.row(), point.column());
+        return point.move(direction);
+    }
+
+    private boolean isPointIsEndLine(Point point) {
+        if (point.column() < ladder.getHeight()) {
```

Comment:
if문을 아래와 같이 한 줄로 표현해볼 수도 있겠군요 😃 

```suggestion
return point.column() >= ladder.getHeight()
```